dbms/src/IO/ReadHelpers.cpp

```diff
@@ -552,9 +552,32 @@ void readCSVStringInto(Vector & s, ReadBuffer & buf, const FormatSettings::CSV &
         while (!buf.eof())
         {
             char * next_pos = buf.position();
-            while (next_pos < buf.buffer().end()
-                && *next_pos != delimiter && *next_pos != '\r' && *next_pos != '\n')    /// NOTE You can make a SIMD version.
-                ++next_pos;
+
+            [&]() {
```

Comment:
Why lambda? Just code block `{...}` is Ok.

Reply:
There is a early return